lib/active_merchant/billing/gateways/komoju.rb

```diff
@@ -51,50 +54,66 @@ def capture(*args)
       end
 
       def store(payment, options = {})
-        params = { payment_details: payment_details(payment, options) }
-        commit("/tokens", params)
+        post = {}
+        add_payment_details(post, payment, options)
+
+        if options[:customer_profile]
+          post[:email] = options[:email]
+          commit("/customers", post)
+        else
+          commit("/tokens", post)
+        end
       end
 
       private
 
-      def payment_details(payment, options)
-        details = if payment.is_a?(CreditCard)
-                    credit_card_payment_details(payment, options)
-                  else
-                    payment
-                  end
-
-        details[:email] = options[:email] if options[:email] && details.is_a?(Hash)
```

Comment:
Seems maybe one difference here is that `options[:email]` is not assigned to `details[:email]` unless details is a Hash. But we're always sending the payment as a hash, so this has no actual impact.